Content/Items/Gear/VanillaItems/Clones/Miscellaneous/SolarEruption.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Terraria.DataStructures;
+using Terraria.ID;
+
+namespace PathOfTerraria.Content.Items.Gear.VanillaItems.Clones.Miscellaneous;
+
+internal class SolarEruption : VanillaClone
+{
+	protected override short VanillaItemId => ItemID.SolarEruption;
+
+	public override void Defaults()
+	{
+		ItemType = Core.ItemType.Melee;
+		base.Defaults();
+	}
+
+	public override bool Shoot(Player player, EntitySource_ItemUse_WithAmmo source, Vector2 position, Vector2 velocity, int type, int damage, float knockback)
+	{
+		float rotation = (Main.rand.NextFloat() - 0.5f) * ((float)Math.PI / 4f);
+		Projectile.NewProjectile(source, position.X, position.Y, velocity.X, velocity.Y, type, damage, knockback, Main.myPlayer, 0f, rotation);
+
+		return false;
+	}
+}
```

Comment:
Just wondering - Why did we need all this custom logic? Do these items actually differ at all?

Reply:
Solar Eruption and Vampire Knives both have extra code that isn't cloned by VanillaClone and has to be manually replicated. In this file, Solar Eruption needs a manually set rotation value when spawned.